ctx.go

```diff
@@ -357,9 +358,56 @@ func (c *netHttpContext[B, P]) Body() (B, error) {
 }
 
 func (c *netHttpContext[B, P]) Params() (P, error) {
-	var p P
+	p := new(P)
 
-	return p, nil
+	paramsType := reflect.TypeOf(p).Elem()
+	if paramsType.Kind() != reflect.Struct {
+		return *p, fmt.Errorf("params must be a struct, got %T", p)
+	}
+	paramsValue := reflect.ValueOf(p).Elem()
+
+	for i := range paramsType.NumField() {
+		field := paramsType.Field(i)
+
+		var paramValue string
+		tag := field.Tag.Get("query")
+		if tag != "" {
+			paramValue = c.QueryParam(tag)
+			if paramValue == "" {
+				continue
+			}
+		} else {
+			tag = field.Tag.Get("header")
+			if tag != "" {
+				paramValue = c.Header(tag)
+				if paramValue == "" {
+					continue
+				}
+			}
+		}
+		fieldValue := paramsValue.Field(i)
+
+		switch field.Type.Kind() {
```

Comment:
Had a colleague point this out to me today. We currently don't support arrays as query params. This is supported in openAPI 3.0 so we should probably support it. 

I personally don't think we should add support it in the non typed params as we should encourage used to use this feature. 

We should also only allow one level. What are your thoughts?

I was beginning to implement this on the non typed styled params and decided to stop once I remembered this work.

Reply:
Done in #501